examples/models/file/latex_extension_mathjax.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from scipy.special import jv
+
+from bokeh.models import Label
+from bokeh.palettes import Spectral4
+from bokeh.core.properties import (
+  Float
+)
+from bokeh.plotting import figure, show
+from bokeh.util.compiler import TypeScript
+
+class LatexLabel(Label):
+    """A subclass of `Label` with additional class attributes 'width' and 'height',
+    canvas mode isn't supported and DOM manipulation happens in the TypeScript
+    superclass implementation that requires setting `render_mode='css'`).
+
+    Only the render method of LabelView is overwritten to perform the
+    text -> latex (via MathJax) conversion
+    """
+
+    width = Float(help="""
+    The width of the rendered label in pixels
+    """)
+    height = Float(help="""
+    The height of the rendered label in pixels
+    """)
+
+    __javascript__ = ["https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js"]
+    __implementation__ = TypeScript("""
+import {Label, LabelView} from "models/annotations/label"
+import * as p from "core/properties"
+
+declare namespace MathJax {
+  function tex2svg(input: string): HTMLElement
+}
+
+export class LatexLabelView extends LabelView {
+  model: LatexLabel
+
+  protected _render(): void {
+    const panel = this.layout != null ? this.layout : this.plot_view.frame
+
+    const xscale = this.coordinates.x_scale
+    const yscale = this.coordinates.y_scale
+
+    let sx = this.model.x_units == "data" ? xscale.compute(this.model.x) : panel.bbox.xview.compute(this.model.x)
+    let sy = this.model.y_units == "data" ? yscale.compute(this.model.y) : panel.bbox.yview.compute(this.model.y)
+
+    sx += this.model.x_offset
+    sy -= this.model.y_offset
+
+    const svg_element = MathJax.tex2svg(this.model.text).children[0] as SVGElement
+    const outer_HTML = svg_element.outerHTML
+    const blob = new Blob([outer_HTML], {type:'image/svg+xml;charset=utf-8'})
+    const blob_URL = URL.createObjectURL(blob)
+
+    const image = new Image()
+
+    image.onload = () => {
+      this.layer.ctx.drawImage(image, sx, sy, this.model.width, this.model.height)
+
+      this.notify_finished()
```

Comment:
So where are we with deciding whether `onload`  triggers immediately in this context? Because if it does trigger immediately, then notification is not needed. If it doesn't, then `drawImage()` needs to be replaced with `request_paint()` with logic preventing infinite repaint loop.

Reply:
On its own, this `notify_finished()` does nothing, as notification was already sent. One has to change `has_finished` method to make sure that notification is not sent before this line. See `ImageURL.has_finished`. This is related to the above point about preventing infinite paint loop.

Reply:
> So where are we with deciding whether onload triggers immediately in this context? 

Is there a reference to a question/issue I can go look at?

Reply:
Earlier in this thread (https://github.com/bokeh/bokeh/pull/11067#discussion_r595576100).

Reply:
Sorry the wording made it sound like you were asking about a policy decision inside our codebase. I am not sure there is any way to know for sure easily. The only reference I could find was from ~10 years ago, and seemed to indicate (from empirical testing) "usually but not always":  https://stackoverflow.com/a/4807114/3406693 But it's possible that things have changed in the interim. Still if it is not documented in a spec somewhere I think we have to play it safe and assume the answer is No

Reply:
> On its own, this notify_finished() does nothing, as notification was already sent. One has to change

working on this